tests/test_avepool2d_dense.py

```diff
@@ -1,241 +1,61 @@
 import numpy as np
 import tensorflow as tf
 import ml_genn as mlg
+import pytest
 from converter import Converter
 
-
-def model_compare_tf_and_mlg(tf_model, x, connectivity_type='procedural'):
-    # Run TensorFlow model
-    tf_y = tf_model(x).numpy()
-
-    # Run ML GeNN model
-    mlg_model = mlg.Model.convert_tf_model(tf_model, converter=Converter(),
-                                           connectivity_type=connectivity_type,
-                                           dt=1.0, batch_size=1)
-    mlg_model.outputs[0].neurons.set_threshold(np.float64(np.inf))
-    mlg_model.set_input_batch(x)
-    mlg_model.step_time(2)
-
-    nrn = mlg_model.outputs[0].neurons.nrn
-    nrn.pull_var_from_device('Vmem')
-    mlg_y = nrn.vars['Vmem'].view.reshape(tf_y.shape)
-
-    assert(np.allclose(mlg_y, tf_y, atol=0.0, rtol=1.0e-3))
-
-    return mlg_model
-
-
-def model_input_0():
-    return np.array([
-        [1, 0, 0, 1, 1, 0, 1, 1, 1, 0],
-        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-        [1, 1, 1, 1, 1, 1, 1, 1, 1, 0],
-        [1, 0, 0, 1, 1, 0, 1, 1, 1, 0],
-        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-        [1, 1, 1, 1, 1, 1, 1, 1, 1, 0],
-        [1, 1, 1, 1, 1, 1, 1, 1, 1, 0],
-        [1, 0, 0, 1, 1, 0, 1, 1, 1, 0],
-        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-    ], dtype=np.float64)
-
-
-def model_input_1():
-    return np.array([
-        [1, 1, 1, 0, 0, 0, 0, 0, 0, 0],
-        [1, 1, 1, 0, 1, 1, 0, 0, 0, 0],
-        [1, 1, 1, 0, 1, 1, 0, 0, 1, 0],
-        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-        [1, 0, 0, 1, 1, 0, 1, 1, 1, 0],
-        [0, 0, 0, 1, 1, 0, 1, 1, 1, 0],
-        [0, 0, 0, 0, 0, 0, 1, 1, 1, 0],
-        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-    ], dtype=np.float64)
-
-
-def test_avepool2d_dense_in_chan_1_padding_valid():
-    '''
-    Test AvePool2DDense with 1 input channel, 1 output channel and valid pool padding.
-    '''
-
+@pytest.mark.parametrize(
+    'in_size, in_chan, out_size, pool_size, pool_strides, connect', [
+        (20, 1, 10, 2, 2, 'sparse'),
+        (20, 1, 10, 2, 2, 'procedural'),
+        (20, 1, 10, 2, 2, 'toeplitz'),
```

Comment:
There isn't a toeplitz implementation of this (as it's basically dense) so probably no point in essentially testing procedural twice

Reply:
Done and added error throw if one tries to use toeplitz with avepool_dense synapses.